pandas/tests/frame/methods/test_set_axis.py

```diff
@@ -32,14 +32,15 @@ def test_set_axis_copy(self, obj):
         result = obj.set_axis(new_index, axis=0, copy=True)
         tm.assert_equal(expected, result)
         assert result is not obj
-        # check we DID make a copy
-        if obj.ndim == 1:
-            assert not tm.shares_memory(result, obj)
-        else:
-            assert not any(
-                tm.shares_memory(result.iloc[:, i], obj.iloc[:, i])
-                for i in range(obj.shape[1])
-            )
+        if not using_copy_on_write:
+            # check we DID make a copy
+            if obj.ndim == 1:
+                assert not tm.shares_memory(result, obj)
+            else:
+                assert not any(
+                    tm.shares_memory(result.iloc[:, i], obj.iloc[:, i])
+                    for i in range(obj.shape[1])
+                )
```

Comment:
why do we need the `if` block here? If `copy=True` was explicitly passed, then wouldn't we expect that memory not be shared regardless of `copy_on_write`?